### PR TITLE
work on components

### PR DESCRIPTION
**Pull request by lionelvillard**
- Bring in parts of Scott's rigging framework
- run `go test --build-images=true` to build and publish images. Default is false
- Add sequence stepper as an component example

### CODE REVIEW

File: pkg/components/sequencestepper/yaml.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sequencestepper
+
+const (
+	podTemplate = `
+apiVersion: v1
```
Comment (slinkydeveloper) [inline pkg/components/sequencestepper/yaml.go:21]: Can we have the templates in separate files?

**Review by slinkydeveloper — COMMENTED** [review]
The sequencestepper image seems exactly the same as the event transformer image in eventing

File: pkg/components/sequencestepper/yaml.go (hunk 1 above)
Comment (lionelvillard) [reply pkg/components/sequencestepper/yaml.go:21]: yes eventually. Can we do this in a different PR?

**Comment by lionelvillard** [comment]
@slinkydeveloper sequencestepper was probably not the best first component to bring over. It's mostly used as an example to realize some of the ideas.

**Comment by slinkydeveloper** [comment]
np @lionelvillard , just let's make sure that while working on the new test framework we don't bring in old bloat :smile:

**Pull request merged**